MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/ElevationQueryActivity.java

```diff
@@ -0,0 +1,285 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textSize;
+
+/**
+ * Use the Mapbox Tilequery API to access global elevation data. More info about
+ * the elevation data can be found at
+ * https://docs.mapbox.com/help/troubleshooting/access-elevation-data/
+ */
+public class ElevationQueryActivity extends AppCompatActivity implements
+  OnMapReadyCallback, PermissionsListener, MapboxMap.OnMapClickListener {
+
+  private static final String RESULT_GEOJSON_SOURCE_ID = "RESULT_GEOJSON_SOURCE_ID";
+  private static final String LAYER_ID = "LAYER_ID";
+  private PermissionsManager permissionsManager;
+  private MapboxMap mapboxMap;
+  private MapView mapView;
+  private TextView elevationQueryNumbersOnlyResponseTextView;
+  private TextView elevationQueryJsonResponseTextView;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_javaservices_elevation_query);
+
+    elevationQueryJsonResponseTextView = findViewById(R.id.elevation_query_api_response_json_textview);
+    elevationQueryNumbersOnlyResponseTextView = findViewById(R.id.elevation_query_api_response_elevation_numbers_only);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @SuppressWarnings( {"MissingPermission"})
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+    ElevationQueryActivity.this.mapboxMap = mapboxMap;
+    mapboxMap.setStyle(Style.OUTDOORS, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        addResultLayer(style);
+        displayDeviceLocation(style);
+        mapboxMap.addOnMapClickListener(ElevationQueryActivity.this);
+        Toast.makeText(ElevationQueryActivity.this, R.string.click_on_map_instruction, Toast.LENGTH_SHORT).show();
+      }
+    });
+  }
+
+  /**
+   * Add a map layer which will show a text number of the highest or lowest elevation number returned
+   * by the Tilequery API.
+   */
+  private void addResultLayer(@NonNull Style loadedMapStyle) {
+    loadedMapStyle.addSource(new GeoJsonSource(RESULT_GEOJSON_SOURCE_ID));
+    loadedMapStyle.addLayer(new SymbolLayer(LAYER_ID, RESULT_GEOJSON_SOURCE_ID).withProperties(
+      textField(get("ele")),
+      textColor(Color.BLUE),
+      textSize(23f),
+      textHaloBlur(10f),
+      textIgnorePlacement(true),
+      textAllowOverlap(true)
+    ));
+  }
+
+
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+    Style style = mapboxMap.getStyle();
```

Comment:
Shall we use the async callback here for the `getStyle()`?

Reply:
Sure. Fixed